app/util/network.py

```diff
@@ -138,45 +136,41 @@ def are_hosts_same(host_a, host_b):
 
         :type host_a: str
         :type host_b: str
-        :return: return False if there was an error getting the rsa key, otherwise returns whether hosts a and b
+        :return: return False if there was an error getting the host id, otherwise returns whether hosts a and b
             are the same host.
         :rtype: bool
         """
-        # For efficiency's sake, skip the rsa key check if the host strings are identical
+        # For efficiency's sake, skip getting host ids if the host strings are identical
         if host_a == host_b and host_a is not None:
             return True
 
-        host_a_rsa_key = Network.rsa_key(host_a)
+        host_a_id = Network.get_host_id(host_a)
 
-        if host_a_rsa_key is None:
+        if host_a_id is None:
             return False
 
-        host_b_rsa_key = Network.rsa_key(host_b)
+        host_b_id = Network.get_host_id(host_b)
 
-        if host_b_rsa_key is None:
+        if host_b_id is None:
             return False
 
-        return host_a_rsa_key == host_b_rsa_key
+        return host_a_id == host_b_id
 
     @staticmethod
-    def rsa_key(host):
+    def get_host_id(host):
         """
-        :param host: The RSA key for host that we want to retrieve
+        :param host: The id for host that we want to retrieve
         :type host: str
-        :return: the rsa key string, without the 'ssh-rsa' prefix. Returns None if failed ssh-keyscan fails.
+        :return: the id of the host
         :rtype: str|None
         """
-        proc = subprocess.Popen('ssh-keyscan -t rsa {}'.format(host), shell=True, stdout=PIPE, stderr=PIPE)
-        output, error = proc.communicate()
-
-        if proc.returncode != 0:
-            log.get_logger(__name__).error('Failed to get rsa string with output: {}, error: {}'.format(output, error))
+        try:
```

Comment:
I'd prefer throwing an error rather than returning None. That way you can handle the error case with a simple try block. Maybe just annotate that it throws a socket.gaierror and handle it in are_host_same